packages/vaadin-grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -96,6 +95,21 @@ export const KeyboardNavigationMixin = (superClass) =>
       this.addEventListener('mouseup', () => (this._isMousedown = false));
     }
 
+    /** @private */
+    get __rowFocusMode() {
+      return this.__isRow(this._itemsFocusable);
```

Comment:
Why does this getter check only whether the `_itemsFocusable` element is a row while `__rowFocusMode` operates also with `_footerFocusable` and `_headerFocusable`?

Shouldn't it also rely on `_footerFocusable` and `_headerFocusable`?

Reply:
In practice, items, footer and header are always in the same focus mode; it's not possible that header is in cell focus mode while footer is in row focus mode.

I was thinking of adding an internal flag for this (`__rowFocusModeValue`), but that felt kind of redundant as the state is already available to be read like this.

Reply:
Now I see, thanks!